packages/module/src/Hooks/pagination.ts

```diff
@@ -1,31 +1,84 @@
-import { useState } from "react";
+import { useState, useEffect } from "react";
 
 export interface UseDataViewPaginationProps {
   /** Initial page */
   page?: number;
   /** Items per page */
   perPage: number;
+  /** Current search parameters as a string */
+  searchParams?: URLSearchParams;
+  /** Function to set search parameters */
+  setSearchParams?: (params: URLSearchParams) => void;
 }
 
 export interface DataViewPaginationProps extends UseDataViewPaginationProps {
   /** Current page number */
-    page: number;
+  page: number;
 }
 
-export const useDataViewPagination = ({ page = 1, perPage }: UseDataViewPaginationProps) => { 
-  const [ state, setState ] = useState({ page, perPage });
-  
-  const onPerPageSelect = (_event: React.MouseEvent | React.KeyboardEvent | MouseEvent | undefined, newPerPage: number) => {
-    setState(prev => ({ ...prev, perPage: newPerPage }));
-  }
-  
-  const onSetPage = (_event: React.MouseEvent | React.KeyboardEvent | MouseEvent | undefined, newPage: number) => {
+export const useDataViewPagination = ({
+  page = 1,
+  perPage,
+  searchParams,
+  setSearchParams,
+}: UseDataViewPaginationProps) => {
+  const [ state, setState ] = useState({
+    page: searchParams?.get('page') !== null
+      ? parseInt(searchParams?.get('page') || `${page}`)
+      : page,
+    perPage: searchParams?.get('perPage') !== null
+      ? parseInt(searchParams?.get('perPage') || `${perPage}`)
+      : perPage,
+  });
+
+  useEffect(() => {
+    if (searchParams && setSearchParams) {
+      const params = new URLSearchParams(searchParams);
+      let updated = false;
+
+      if (!params.has('page')) {
+        params.set('page', `${page}`);
+        updated = true;
+      }
+
+      if (!params.has('perPage')) {
+        params.set('perPage', `${perPage}`);
+        updated = true;
+      }
+
+      updated && setSearchParams(params);
+    }
+  // eslint-disable-next-line react-hooks/exhaustive-deps
+  }, []);
```

Comment:
I assume we are deliberately ignoring the changes in the inputs? But what if the URL changes by other means that just through the hook? Should the component react? Maybe the query values themselves should be in the dependency array.